pints/tests/test_error_measures.py

```diff
@@ -108,25 +115,52 @@ def test_mean_squared_error_single(self):
         # Set up problem
         model = pints.toy.ConstantModel(1)
         times = [1, 2, 3]
+
+        # Test Case I: Input as List
         values = [1, 1, 1]
         p = pints.SingleOutputProblem(model, times, values)
 
-        # Test
+        # Test for different parameters: expected = mean(input - 1) ** 2
+        e = pints.MeanSquaredError(p)
+        self.assertEqual(e.n_parameters(), 1)
+        self.assertEqual(e([1]), 0)
+        self.assertEqual(e([3]), 4)
+
+        # Derivative of error for different parameters:
+        # expected = 2 * mean(input - 1)
+        x = 1
+        y, dy = e.evaluateS1([x])
+        self.assertEqual(y, e([x]))
+        self.assertEqual(dy.shape, (1,))
+        self.assertEqual(dy, 2 * (x - 1))
+        x = 3
+        y, dy = e.evaluateS1([x])
+        self.assertEqual(y, e([x]))
+        self.assertEqual(dy.shape, (1,))
+        self.assertEqual(dy, 2 * (x - 1))
+
+        # Test Case II: Input as array of shape (n_times, 1)
```

Comment:
"Test Case II" is a good example of where it looks like you could split into two methods

Reply:
Ok. now that I think about it. Does it maybe make sense to split the sensitivity checks? I think this would make the tests much smaller.

Reply:
Sure!

I was also thinking about trying out [this](https://github.com/pints-team/pints/issues/1172) at some point...

Reply:
Ok I'll give it a go for one likelihood and then let you have a look

Reply:
@Michael 
Whenever you have time, I refactored the ScaledLogLikelihood test. I personally think it's a lot tidier now than before, but let me know what you think. We could try to implement the other tests in this pattern too.

Reply:
That looks great!